.github/workflows/lint.yml

```diff
@@ -0,0 +1,24 @@
+---
+# SPDX-license-identifier: Apache-2.0
```

Comment:
I believe we need this to follow the existing license format (see other yaml files). This means that instead of SPDX license we need the Copyright to CHIP authors (had a long discussion on this and unfortunatleyt the result was the short version was not sufficient)